File: event-store/sdks/sdk-py/examples/basic.py
import os
from sdk_py.client_rt import EventStoreClientRT

addr = os.environ.get("EVENTSTORE_ADDR", "localhost:50051")
client = EventStoreClientRT(addr)

append_resp = client.append({
    "aggregate_id": "Order-PY-1",
    "aggregate_type": "Order",
    "expected_any": "NO_AGGREGATE",
    "events": [ { "meta": { "event_type": "OrderCreated" }, "payload": "aGVsbG8=" } ]
})
print("append:", append_resp)

read_resp = client.read_stream({ "aggregate_id": "Order-PY-1", "from_aggregate_nonce": 1, "max_count": 100, "forward": True })
print("read:", read_resp)

File: event-store/sdks/sdk-ts/src/examples/basic.ts
import { EventStoreClientRT } from "../runtime-client";

async function main() {
  const addr = process.env.EVENTSTORE_ADDR ?? "localhost:50051";
  const client = new EventStoreClientRT(addr);

  const appendResp = await client.append({
    aggregate_id: "Order-TS-1",
    aggregate_type: "Order",
    expected_any: "ANY",
    events: [
      {
        meta: {
          aggregate_nonce: 1,  // client proposes the nonce
          event_type: "OrderCreated"
        },
        payload: Buffer.from("hello").toString("base64")
      },
    ],
  });
  console.log("append:", appendResp);

  const readResp = await client.readStream({
    aggregate_id: "Order-TS-1",
    from_aggregate_nonce: 1,
    max_count: 100,
    forward: true
  });
  console.log("read:", readResp);
}

main().catch((e) => { console.error(e); process.exit(1); });
